packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -396,22 +428,117 @@ abstract class ImageStreamCompleter with Diagnosticable {
     }
   }
 
+  /// Adds a listener callback that is called whenever a new concrete [ImageInfo]
+  /// object is available or an error is reported. If a concrete image is
+  /// already available, or if an error has been already reported, this object
+  /// will notify the listener synchronously.
+  ///
```

Comment:
This should document that listeners are in charge of disposing the image they receive.

Reply:
Added to template.